File: CS/WindowsFormsApplication1/FixedItemCreator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;

namespace WindowsFormsApplication1
{
    public class FixedItemCreator
    {
        List<BarItem> fixedItems = new List<BarItem>();
        RibbonControl Ribbon;

        public FixedItemCreator(RibbonControl ribbon) 
        {
            fixedItems = new List<BarItem>();
            Ribbon = ribbon;
            Ribbon.SelectedPageChanged += new EventHandler(Ribbon_SelectedPageChanged);
        }

        void Ribbon_SelectedPageChanged(object sender, EventArgs e)
        {
            List<BarItem> absentItems = new List<BarItem>();
            foreach (BarItem fixedItem in fixedItems)
            {
                bool founded = false;
                foreach (RibbonPageGroup group in Ribbon.SelectedPage.Groups)
                {

                    foreach (BarItemLink itemLink in group.ItemLinks)
                    {
                        if (itemLink.Item == fixedItem)
                        {
                            founded = true;
                        }

                    }
                }
                if (!founded)
                {
                    absentItems.Add(fixedItem);
                }
            }

            foreach (BarItem absentItem in absentItems)
            {
                RibbonPageGroup patternGroup = (absentItem.Links[0].LinkedObject as RibbonPageGroupItemLinkCollection).PageGroup;
                RibbonPageGroup currentGroup = null;
                foreach (RibbonPageGroup group in Ribbon.SelectedPage.Groups)
                {
                    if (group.Text == patternGroup.Text)
                    {
                        currentGroup = group;
                        break;
                    }
                }
                if (currentGroup == null)
                {
                    currentGroup = new RibbonPageGroup();
                    currentGroup.Text = patternGroup.Text;
                    if (patternGroup.Page.Groups.IndexOf(patternGroup) >= Ribbon.SelectedPage.Groups.Count)
                    {
                        Ribbon.SelectedPage.Groups.Add(currentGroup);
                    }
                    else
                    {
                        Ribbon.SelectedPage.Groups.Insert(patternGroup.Page.Groups.IndexOf(patternGroup), currentGroup);
                    }
                    
                }
                currentGroup.ItemLinks.Add(absentItem);
            }
        }

        public void AddItem(BarItem item)
        {
            if (!fixedItems.Contains(item))
            {
                fixedItems.Add(item);
            }        
        }
    }
}

File: VB/WindowsFormsApplication1/FixedItemCreator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon

Namespace WindowsFormsApplication1
	Public Class FixedItemCreator
		Private fixedItems As New List(Of BarItem)()
		Private Ribbon As RibbonControl

		Public Sub New(ByVal ribbon As RibbonControl)
			fixedItems = New List(Of BarItem)()
			Me.Ribbon = ribbon
			AddHandler Me.Ribbon.SelectedPageChanged, AddressOf Ribbon_SelectedPageChanged
		End Sub

		Private Sub Ribbon_SelectedPageChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim absentItems As New List(Of BarItem)()
			For Each fixedItem As BarItem In fixedItems
				Dim founded As Boolean = False
				For Each group As RibbonPageGroup In Ribbon.SelectedPage.Groups

					For Each itemLink As BarItemLink In group.ItemLinks
                        If itemLink.Item.Equals(fixedItem) Then
                            founded = True
                        End If

					Next itemLink
				Next group
				If (Not founded) Then
					absentItems.Add(fixedItem)
				End If
			Next fixedItem

			For Each absentItem As BarItem In absentItems
				Dim patternGroup As RibbonPageGroup = (TryCast(absentItem.Links(0).LinkedObject, RibbonPageGroupItemLinkCollection)).PageGroup
				Dim currentGroup As RibbonPageGroup = Nothing
				For Each group As RibbonPageGroup In Ribbon.SelectedPage.Groups
					If group.Text = patternGroup.Text Then
						currentGroup = group
						Exit For
					End If
				Next group
				If currentGroup Is Nothing Then
					currentGroup = New RibbonPageGroup()
					currentGroup.Text = patternGroup.Text
					If patternGroup.Page.Groups.IndexOf(patternGroup) >= Ribbon.SelectedPage.Groups.Count Then
						Ribbon.SelectedPage.Groups.Add(currentGroup)
					Else
						Ribbon.SelectedPage.Groups.Insert(patternGroup.Page.Groups.IndexOf(patternGroup), currentGroup)
					End If

				End If
				currentGroup.ItemLinks.Add(absentItem)
			Next absentItem
		End Sub

		Public Sub AddItem(ByVal item As BarItem)
			If (Not fixedItems.Contains(item)) Then
				fixedItems.Add(item)
			End If
		End Sub
	End Class
End Namespace
